lib/codecs/src/encoding/format/csv.rs

```diff
@@ -392,37 +410,89 @@ mod tests {
     #[test]
     fn correct_quoting() {
         let event = Event::Log(LogEvent::from(btreemap! {
-            "field1" => Value::from(1),
-            "field2" => Value::from("foo\"bar"),
+            "field1" => Value::from("hello world"),
+            "field2" => Value::from(1),
+            "field3" => Value::from("foo\"bar"),
+            "field4" => Value::from("baz,bas"),
         }));
         let fields = vec![
             ConfigTargetPath::try_from("field1".to_string()).unwrap(),
             ConfigTargetPath::try_from("field2".to_string()).unwrap(),
+            ConfigTargetPath::try_from("field3".to_string()).unwrap(),
+            ConfigTargetPath::try_from("field4".to_string()).unwrap(),
         ];
-        let opts = CsvSerializerOptions {
-            fields,
+
+        let mut default_bytes = BytesMut::new();
+        let mut never_bytes = BytesMut::new();
+        let mut always_bytes = BytesMut::new();
+        let mut non_numeric_bytes = BytesMut::new();
+
+        let mut default_serializer = CsvSerializerConfig::new(CsvSerializerOptions {
```

Comment:
💬 To enhance readability please group the `default_serializer` and `default_bytes` statements together. Same for the other serializers.

Reply:
Yes, makes sense to write multiple lines to the buffer. Will do that tomorrow.
In regards to the ordering within the test I actually hat your suggested arrangement first and felt like having a visible Arrange Act Assert pattern makes it more readable. Still I have no strong feelings about it so changing it back is fine by me.

Reply:
[3512a7a](https://github.com/vectordotdev/vector/pull/18320/commits/3512a7a43e232881f956c7232502a97b42490854) did the mentioned refactor in the quoting test and adds the writing of to lines to one buffer which simply concatenates the two values. I am assuming in a real live example those would be concatenated in the framing part of vector. Is the test what you where intending @pront ?

Reply:
Looks good now. Thanks for updating.